### PR TITLE
feat(metadata): Generically display metadata

### LINKED ISSUE #697: Support viewing metadata locations
### Description

As a catalog author, I want to view locations in regard to metadata.

### Acceptance Criteria

* The `location`s field and it's respective metadata is properly displayed within the React library. A `location`s metadata include:
   * `uuid` (not displayed to author)
   * `title` [1]
   * `address` [1]
   * `email-address` [0 to many)
   * `telephone-number` [0 to many)
   * `url` [0 to many)
   * `prop` [0 to many)
   * `link` [0 to many)
   * `remarks` [0 to many)

### Proposed Solution

_No response_

### Additional Notes

A lot of this should already be support. Some of a location's metadata like `link` and `remarks` are not currently supported.

### Implementation

- [ ] I can (or plan to) submit a pull request to implement this
- [ ] Implementing this may result in a breaking change

### LINKED ISSUE #702: Support viewing metadata `published`
### Description

As a catalog author, I want to view `published` in regard to metadata.

### Acceptance Criteria

* The `published` field is properly displayed within the React library

### Proposed Solution

This could be placed above `oscal-version` in the panel to the right of parties

### Additional Notes

_No response_

### Implementation

- [ ] I can (or plan to) submit a pull request to implement this
- [ ] Implementing this may result in a breaking change

### PR DESCRIPTION
**Pull request by tuckerzp**
[OSCAL metadata](https://pages.nist.gov/OSCAL/reference/latest/complete/json-reference/#/catalog/metadata) contains a few list items we have been displaying
using material ui Cards. These include `parties`, `roles`, and now
`locations`. 

While the data for these items differ, how we display them is very
similar. This will make displaying lists in metadata using Cards
a lot more generic.

Closes #697
Closes #702

### CODE REVIEW

**Comment by tuckerzp** [comment]
@kylelaker & @Bronstrom I tried to create a draft as quickly as possible. I did not want to go too deep into the rabbit hole of messing with the `Cards` if I was very off on my approach. I had a few things I would love your thoughts on. 

Currently the overall code structure has gotten cleaner (I think), but the structure of how its presented has suffered. 

1. Overall cleanness of the code. The `metadata` lists differ quite a bit, I tried to make the `Sections` & `Cards` be as generic as possible without getting too crazy with it.
2. Making things look less terrible. The `Grid` is a powerful thing that I still am working on understanding. I am not completely sure how to make the `AdditionalMetadata` section look nice without comletely rewriting it. I am wondering if it is some small thing with the `Grids` that I am missing. I also feel like the `alignitems=stretch` makes the cards look weird when there are only a few of them. 

image

**Comment by Bronstrom** [comment]
> 1. Overall cleanness of the code. The metadata lists differ quite a bit, I tried to make the Sections & Cards be as generic as possible without getting too crazy with it.

I like the structural improvements made here and it would nice to keep them or iterate on them.

> 2. Making things look less terrible. The Grid is a powerful thing that I still am working on understanding. I am not completely sure how to make the AdditionalMetadata section look nice without comletely rewriting it. I am wondering if it is some small thing with the Grids that I am missing. I also feel like the alignitems=stretch makes the cards look weird when there are only a few of them.

Here are a couple of design suggests I don't believe will not require a whole rework:
* The additional Metadata section (currently the `Version`, `Last Modified`, and `OSCAL Version` section), has a lot of negative space around it that seems unnecessary. I would suggest trying to make use of a similar Grid design pattern here to the Parties and Roles sections, where these elements appear on the same row vertically. For now, these rows can have 3 elements in them, with only three elements currently supported for this section, but it may make sense to have 4/5 in a row later. 
*  Secondly, I would try to add some sort of border to the Grids, and potentially add some more space around the Heading for "Parties" and "Roles". I'm not sure if the `Card` supports some sort of inner border to add for the whole "Metadata pane", but adding the CSS property `justify-content: space-around` would probably work in the case of spacing the grids out.

Not to confuse anyone on this being an official implementation, but this is a quick design mock-up I "photoshopped" in MS Paint:

image

**Comment by kylelaker** [comment]
> 2\. without comletely rewriting it

It is basic enough that I am not *opposed* to this idea if it accomplishes your goals here better.

**Comment by kylelaker** [comment]
> * The additional Metadata section (currently the `Version`, `Last Modified`, and `OSCAL Version` section), has a lot of negative space around it that seems unnecessary

Keep in mind that there's more data we want to display in here!! _At least_ the `published` fields and maybe the count of objects in the `revisions` field. These three are not the only three that will ever exist.

**Review by kylelaker — COMMENTED** [review]
AMAZING start! Mostly I am not in love with *naming* which isn't an issue in a first-stab draft PR where I know you're going to be working on this soon anyway. Don't let any of these comments slow you down.

Also, because you know that you'll be using them soon enough, make sure the components for phone number, address, and email are generic enough for what you need!

**Comment by tuckerzp** [comment]
Update after redesign of basic metadata

image

**Comment by tuckerzp** [comment]
Update with Accordion

image

---

image

**Review by kylelaker — APPROVED** [review]
This is a really great refactoring of how we display metadata and I think it will be pretty extensible going forward. There is a lot here but overall it looks good.

I think any remaining comments are either trivial or not necessary so I am going to move forward with approving, but please make sure to resolve the text issue before this gets merged.

File: src/components/OSCALMetadata.js [hunk 1]
```diff
@@ -393,9 +478,146 @@ export default function OSCALMetadata(props) {
...
+      {urls?.length ? (
+        urls.map((url) => (
+          <TextWithIcon key={url} icon={<OpenInNewIcon />}>
+            <Link href={url} target="_blank" rel="noreferrer">
+              {url}
+            </Link>
+          </TextWithIcon>
+        ))
+      ) : (
+        <Typography>So sad! No URLs.</Typography>
```
[the remark this answers is not in the document]
Comment (tuckerzp) [reply src/components/OSCALMetadata.js]: Good catch! Updated.

**Review by Bronstrom — APPROVED** [review]
Overall, this looks great! The structural improvements will make the code more maintainable. The new tests and test data look good. Also, the collapsible metadata sections are pretty clean.

Nice work @tuckerzp!

**Pull request merged**